source/backend/tests/unit/api/Services/AcquisitionFileServiceTest.cs

```diff
@@ -1219,8 +1219,126 @@ public void AddCompensationsRequisitions_Success()
+        {
+            // Arrange
+            var service = CreateAcquisitionServiceWithPermissions();
+
+            // Act
+            Action act = () => service.GetInterestHolders(1);
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+        }
+
+        [Fact]
+        public void GetInterestHolders_NotAuthorized_Contractor()
+        {
+            // Arrange
+            var service = CreateAcquisitionServiceWithPermissions(Permissions.AcquisitionFileView, Permissions.CompensationRequisitionView);
+
+            var userRepository = _helper.GetService<Mock<IUserRepository>>();
+            var contractorUser = EntityHelper.CreateUser(1, Guid.NewGuid(), username: "Test", isContractor: true);
+            userRepository.Setup(x => x.GetUserInfoByKeycloakUserId(It.IsAny<Guid>())).Returns(contractorUser);
+
+            var acqFileRepository = _helper.GetService<Mock<IAcquisitionFileRepository>>();
+            acqFileRepository.Setup(x => x.GetById(It.IsAny<long>())).Returns(EntityHelper.CreateAcquisitionFile());
+
+            // Act
+            Action act = () => service.GetInterestHolders(1);
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+        }
+
+        [Fact]
+        public void GetInterestHolders_Success()
+        {
+            // Arrange
+            var service = CreateAcquisitionServiceWithPermissions(Permissions.AcquisitionFileView, Permissions.CompensationRequisitionView);
+
+            var repository = _helper.GetService<Mock<IInterestHolderRepository>>();
+            repository.Setup(x => x.GetInterestHoldersByAcquisitionFile(It.IsAny<long>()))
+                .Returns(new List<PimsInterestHolder>()
+                {
+                    new PimsInterestHolder(),
+                });
+
+            var userRepository = _helper.GetService<Mock<IUserRepository>>();
+            userRepository.Setup(x => x.GetUserInfoByKeycloakUserId(It.IsAny<Guid>())).Returns(EntityHelper.CreateUser("Test"));
+
+            // Act
+            var result = service.GetInterestHolders(1);
+
+            // Assert
+            repository.Verify(x => x.GetInterestHoldersByAcquisitionFile(It.IsAny<long>()), Times.Once);
+        }
 
+        [Fact]
+        public void UpdateInterestHolders_NoPermissions()
+        {
+            // Arrange
+            var service = CreateAcquisitionServiceWithPermissions();
+
+            // Act
+            Action act = () => service.UpdateInterestHolders(1, new List<PimsInterestHolder>()
+                {
+                    new PimsInterestHolder(),
+                });
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+        }
 
+        [Fact]
+        public void UpdateInterestHolders_NotAuthorized_Contractor()
+        {
+            // Arrange
+            var service = CreateAcquisitionServiceWithPermissions(Permissions.CompensationRequisitionAdd);
```

Comment:
please change this permission to AcquisitionFileEdit so the contractor path is actually tested. As it is right now, the check for AcquisitionFileEdit claim runs first and throws the exception before checking for contractor status